Subject: Misc KDB fixes for RT (debug builds)
From: Tony Jones <tonyj@suse.de>
Upstream: no

- Switch to usiing mapping_nrpages
- Use lock_list next/prev members

---
 kdb/modules/kdbm_pg.c |    2 +-
 kdb/modules/kdbm_vm.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.27-RT/kdb/modules/kdbm_pg.c
===================================================================
--- linux-2.6.27-RT.orig/kdb/modules/kdbm_pg.c
+++ linux-2.6.27-RT/kdb/modules/kdbm_pg.c
@@ -491,7 +491,7 @@ kdbm_inode(int argc, const char **argv)
 	kdb_printf(" i_sb = 0x%p i_op = 0x%p i_data = 0x%lx nrpages = %lu\n",
 					inode->i_sb, inode->i_op,
 					addr + offsetof(struct inode, i_data),
-					inode->i_data.nrpages);
+					mapping_nrpages(&inode->i_data));
 	kdb_printf(" i_fop= 0x%p i_flock = 0x%p i_mapping = 0x%p\n",
 			   inode->i_fop, inode->i_flock, inode->i_mapping);
 
Index: linux-2.6.27-RT/kdb/modules/kdbm_vm.c
===================================================================
--- linux-2.6.27-RT.orig/kdb/modules/kdbm_vm.c
+++ linux-2.6.27-RT/kdb/modules/kdbm_vm.c
@@ -741,7 +741,7 @@ kdbm_filp(int argc, const char **argv)
 	kdb_printf("File Pointer at 0x%lx\n", addr);
 
 	kdb_printf(" fu_list.nxt = 0x%p fu_list.prv = 0x%p\n",
-					f.f_u.fu_list.next, f.f_u.fu_list.prev);
+					f.f_u.fu_llist.next, f.f_u.fu_llist.prev);
 
 	kdb_printf(" f_dentry = 0x%p f_vfsmnt = 0x%p f_op = 0x%p\n",
 					f.f_dentry, f.f_vfsmnt, f.f_op);
